AK/URL.cpp

```diff
@@ -116,31 +116,39 @@ void URL::set_fragment(const String& fragment)
     m_fragment = fragment;
 }
 
+// FIXME: This is by no means complete.
+// NOTE: This relies on some assumptions about how the spec-defined URL parser works that may turn out to be wrong.
```

Comment:
This is definitely and idea for the future, but I think for this we need to optimize the parser a bit. In the meantime, I have changed the parser to not use the setters of the `URL` class (since this checks validity and would lead to infinite recursion if we implemented your suggestion). Also, I have added a comment in the parser to explain that the setters should not be used due to this possibility (and the performance overhead of computing the validity all the time even though we set it to `true` at the end anyway).